src/subscription/__tests__/subscribe-test.js

```diff
@@ -211,4 +216,279 @@ describe('Subscribe', () => {
+            inbox: {
+              unread: 1,
+              total: 2,
+            },
+          },
+        },
+      },
+    });
+
+    payload = subscription.next();
+    subscription.return();
+
+    // A new email arrives!
+    expect(sendImportantEmail({
+      from: 'yuzhi@graphql.org',
+      subject: 'Alright 2',
+      message: 'Tests are good 2',
+      unread: true,
+    })).to.equal(false);
+
+    expect(await payload).to.deep.equal({
+      done: true,
+      value: undefined,
+    });
+  });
+
+  it('events order is correct when multiple triggered together', async () => {
+    const pubsub = new EventEmitter();
+    const { sendImportantEmail, subscription } = createSubscription(pubsub);
+    let payload = subscription.next();
+
+    // A new email arrives!
+    expect(sendImportantEmail({
+      from: 'yuzhi@graphql.org',
+      subject: 'Message',
+      message: 'Tests are good',
+      unread: true,
+    })).to.equal(true);
+
+    // A new email arrives!
+    expect(sendImportantEmail({
+      from: 'yuzhi@graphql.org',
+      subject: 'Message 2',
+      message: 'Tests are good 2',
+      unread: true,
+    })).to.equal(true);
+
+    expect(await payload).to.deep.equal({
+      done: false,
+      value: {
+        data: {
+          importantEmail: {
+            email: {
+              from: 'yuzhi@graphql.org',
+              subject: 'Message',
+            },
+            inbox: {
+              unread: 2,
+              total: 3,
+            },
+          },
+        },
+      },
+    });
+
+    payload = subscription.next();
+
+    expect(await payload).to.deep.equal({
+      done: false,
+      value: {
+        data: {
+          importantEmail: {
+            email: {
+              from: 'yuzhi@graphql.org',
+              subject: 'Message 2',
+            },
+            inbox: {
+              unread: 2,
+              total: 3,
+            },
+          },
+        },
+      },
+    });
+  });
+
+  it('invalid query should result in error', async () => {
+    const invalidAST = parse(`
+      subscription {
+        invalidField
+      }
+    `);
+
+    expect(() => {
+      subscribe(
+        emailSchema,
+        invalidAST,
+        null,
+        null, // context
+        { priority: 1 });
+    }).to.throw('This subscription is not defined by the schema.');
+  });
+
+  it('throws when subscription definition doesnt return iterator', () => {
+    const invalidEmailSchema = new GraphQLSchema({
+      query: QueryType,
+      subscription: new GraphQLObjectType({
+        name: 'Subscription',
+        fields: {
+          importantEmail: {
+            type: GraphQLString,
+            resolve: () => 'test',
+          },
+        }
+      })
+    });
+
+    const ast = parse(`
+      subscription {
+        importantEmail
+      }
+    `);
+
+    expect(() => {
+      subscribe(
+        invalidEmailSchema,
+        ast,
+        null,
+        null, // context
+        { priority: 1 });
```

Comment:
note: all the `{priority: 1}` aren't being used by the test AST (those are variables), so these tests could all be simplified to `subscribe(schema, ast)`